### PR TITLE
fixup! ASoC: SOF: Core: Add remove_late() to  sof_init_environment fa…

### PR DESCRIPTION
**Pull request by ujfalusi**
…ilure path

sof_select_ipc_and_paths() error needs to be handled.

### CODE REVIEW

**Review by ckborah — APPROVED** [review]
Thank you @ujfalusi for catching this!

File: sound/soc/sof/core.c [hunk 1]
```diff
@@ -350,7 +350,9 @@ static int sof_init_environment(struct snd_sof_dev *sdev)
 	}
 
 	ret = sof_select_ipc_and_paths(sdev);
-	if (!ret && plat_data->ipc_type != base_profile->ipc_type) {
+	if (ret) {
+		goto err_machine_check;
+	} else if (plat_data->ipc_type != base_profile->ipc_type) {
```
Comment (plbossart) [inline sound/soc/sof/core.c:355]: no longer a fixup, the initial patch was applied by Mark Brown

Comment (ujfalusi) [reply sound/soc/sof/core.c:355]: oh, OK, then I will reword on Monday

**Comment by plbossart** [comment]
@ujfalusi ping? We need to send this upstream.

**Comment by ujfalusi** [comment]
Changes since v1:
- Converted to a proper patch with Fixes tag, commit message reworded accordingly.

**Review by plbossart — APPROVED** [review]

**Pull request merged**